### ISSUE TITLE
NoSuchMethodError: The method 'toDouble' was called on null.

### ISSUE THREAD

**Issue by uzuki-P** [marcelgarus/marquee#17]
Error Log:
```
05-16 14:54:38.913 11877-12010/com.lunata.dating E/flutter: [ERROR:flutter/lib/ui/ui_dart_state.cc(148)] Unhandled Exception: NoSuchMethodError: The method 'toDouble' was called on null.
    Receiver: null
    Tried calling: toDouble()
    #0      Object.noSuchMethod (dart:core-patch/object_patch.dart:50:5)
    #1      double.> (dart:core-patch/double.dart:87:31)
    #2      ScrollPosition._updateSemanticActions (package:flutter/src/widgets/scroll_position.dart:436:16)
    #3      ScrollPosition.notifyListeners (package:flutter/src/widgets/scroll_position.dart:695:5)
    #4      ScrollPosition.forcePixels (package:flutter/src/widgets/scroll_position.dart:318:5)
    #5      ScrollPositionWithSingleContext.jumpTo (package:flutter/src/widgets/scroll_position_with_single_context.dart:201:7)
    #6      ScrollController.jumpTo (package:flutter/src/widgets/scroll_controller.dart:172:16)
    #7      _MarqueeState._makeRoundTrip (package:marquee/marquee.dart:512:17)
    <asynchronous suspension>
    #8      _MarqueeState._scroll (package:marquee/marquee.dart:446:11)
    <asynchronous suspension>
    #9      Future.doWhile.<anonymous closure> (dart:async/future.dart:534:26)
    #10     _rootRunUnary (dart:async/zone.dart:1132:38)
    #11     _CustomZone.runUnary (dart:async/zone.dart:1029:19)
    #12     _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
    #13     _CustomZone.bindUnaryCallbackGuarded.<anonymous closure> (dart:async/zone.dart:968:26)
    #14     Future.doWhile (dart:async/future.dart:549:18)
    #15     _MarqueeState.initState.<anonymous closure> (package:marquee/marquee.dart:441:14)
    #16     new Future.delayed.<anonymous closure> (dart:async/future.dart:316:39)
    #17     _rootRun (dart:async/zone.dart:1120:38)
    #18     _CustomZone.run (dart:async/zone.dart:1021:19)
    #19     _CustomZone.runGuarded (dart:async/zone.dart:923:7)
    #20     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:963:23)
    #21     _rootRun (dart:async/zone.dart:1124:13)
    #22     _CustomZone.run (dart:async/zone.dart:1021:19)
    #23     _CustomZone.bindCallback.<anonymous closure> (dart:async/zone.dart:947:23)
    #24     Timer._createTimer.<anonymous closure> (dart:async-patch/timer_patch.dart:21:15)
    #25     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:382:19)
    #26     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:416:5)
    #27     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:171:12)
```

The code I use is your example:
```
Marquee(
          text: 'Some sample text that takes some space.',
          style: TextStyle(fontWeight: FontWeight.bold),
          scrollAxis: Axis.horizontal,
          crossAxisAlignment: CrossAxisAlignment.start,
          blankSpace: 20.0,
          velocity: 100.0,
          pauseAfterRound: Duration(seconds: 1),
          startPadding: 10.0,
          accelerationDuration: Duration(seconds: 1),
          accelerationCurve: Curves.linear,
          decelerationDuration: Duration(milliseconds: 500),
          decelerationCurve: Curves.easeOut,
 )
```

Flutter version:
```
[√] Flutter (Channel beta, v1.5.4-hotfix.2, on Microsoft Windows [Version 10.0.17763.503], locale en-US)
    • Flutter version 1.5.4-hotfix.2 at J:\program\flutter
    • Framework revision 7a4c33425d (2 weeks ago), 2019-04-29 11:05:24 -0700
    • Engine revision 52c7a1e849
    • Dart version 2.3.0 (build 2.3.0-dev.0.5 a1668566e5)

[√] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
    • Android SDK at J:\android\sdk
    • Android NDK location not configured (optional; useful for native profiling support)
    • Platform android-28, build-tools 28.0.3
    • ANDROID_HOME = J:\android\sdk
    • Java binary at: C:\Program Files\Android\Android Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)
    • All Android licenses accepted.

[√] Android Studio (version 3.4)
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin version 35.3.1
    • Dart plugin version 183.6270
    • Java version OpenJDK Runtime Environment (build 1.8.0_152-release-1343-b01)

[√] IntelliJ IDEA Community Edition (version 2019.1)
    • IntelliJ at C:\Program Files\JetBrains\IntelliJ IDEA Community Edition 2019.1
    • Flutter plugin version 35.3.3
    • Dart plugin version 191.7141.49

[√] VS Code (version 1.33.1)
    • VS Code at C:\Users\Fajar\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.0.2

[√] Connected device (1 available)
    • SM G532G • 4200c83dceeb64b7 • android-arm • Android 6.0.1 (API 23)

• No issues found!
```

**Comment by marcelgarus** [comment]
Thanks for reporting the issue 😄 I'll look into it

**Comment by marcelgarus** [comment]
I reproduced the bug and it seems the problem is this piece of code used internally:
```dart
Future.delayed(Duration(seconds: 1), () {
  Future.doWhile(_scroll);
});
```

It's basically a workaround to let the widget scroll after the first build. Apparently, the Flutter framework changed how it works so this piece is executed *before* the first build. Gladly, I discovered a wonderful new API that solves this problem by hooking into the widget lifecycle directly:
```dart
WidgetsBinding.instance.addPostFrameCallback((_) {
  if (!_running) {
    _running = true;
    Future.doWhile(_scroll);
  }
});
```

The new version is published to pub.dev, so enjoy!
